### PR TITLE
Feature/#21 moshi

### LINKED ISSUE #21: JSON Serialisation Library
Should we use Jackson or try a different one?

### PR DESCRIPTION
**Pull request by aanciaes**
Closes #21

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -22,6 +22,7 @@ apply plugin: 'application'
 apply plugin: 'koin'
 apply plugin: "com.github.johnrengelman.shadow"
 apply plugin: 'org.jlleitschuh.gradle.ktlint'
+apply plugin: 'kotlin-kapt'
```
Comment (felix19350) [inline build.gradle:25]: @aanciaes what is this?

File: src/main/kotlin/test/ktortemplate/core/httphandler/DefaultRoutes.kt [hunk 2]
```diff
@@ -26,12 +31,19 @@ fun Route.defaultRoutes() {
         val pageRequest = call.parsePageRequest()
         val totalElements = carService.count(pageRequest)
         val data = carService.list(pageRequest)
+
+        val type: Type = Types.newParameterizedType(PageResponse::class.java, Car::class.java)
```
Comment (felix19350) [inline src/main/kotlin/test/ktortemplate/core/httphandler/DefaultRoutes.kt:35]: Meh... boilerplate :(

**Review by felix19350 — COMMENTED** [review]
Compared to jackson:
- It deals better with sealed classes
- Has a bit of boilerplate around generics

@aanciaes / @monxalo any more arguments in favor or against moshi?

File: build.gradle (hunk 1 above)
Comment (monxalo) [reply build.gradle:25]: This is a utility for annotation processors for kotlin, since moshi uses this for generating adapters / serializers.

**Pull request closed without merging**